pkg/controllers/corecluster/corecluster_controller_test.go

```diff
@@ -0,0 +1,413 @@
+		Expect(cl.Create(context.Background(), namespace)).To(Succeed())
+		testNamespaceName = namespace.Name
+
+		By("creating the testing OCP infrastructure on AWS")
+		openshiftInfrastructure = configv1resourcebuilder.Infrastructure().AsAWS(testInfraName, testRegionName).Build()
+		Expect(cl.Create(ctx, openshiftInfrastructure)).To(Succeed())
+
+		openshiftInfrastructureCopy = openshiftInfrastructure.DeepCopy()
+		openshiftInfrastructureCopy.Status = configv1.InfrastructureStatus{
+			APIServerInternalURL: "https://test:8081",
+			InfrastructureName:   testInfraName,
+			Platform:             configv1.AWSPlatformType,
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: configv1.AWSPlatformType,
+				AWS: &configv1.AWSPlatformStatus{
+					Region: testRegionName,
+				},
+			},
+		}
+		Expect(cl.Status().Patch(ctx, openshiftInfrastructure, client.MergeFrom(openshiftInfrastructureCopy))).To(Succeed())
+
+		r = &CoreClusterReconciler{
+			ClusterOperatorStatusClient: operatorstatus.ClusterOperatorStatusClient{
+				Client:           cl,
+				ManagedNamespace: testNamespaceName,
+			},
+			Cluster:  &capiv1.Cluster{},
+			Infra:    openshiftInfrastructureCopy.DeepCopy(),
+			Platform: openshiftInfrastructureCopy.Status.PlatformStatus.Type,
+		}
+		Expect(r.SetupWithManager(mgr)).To(Succeed(),
+			"Reconciler should be able to setup with manager")
+
+		By("Starting the manager")
+		mgrCancel, mgrDone = startManager(&mgr)
+	})
+
+	AfterEach(func() {
+		By("Stopping the manager")
+
+		stopManager()
+		Eventually(mgrDone, timeout).Should(BeClosed())
+
+		testutils.CleanupResources(Default, ctx, testEnv.Config, cl, testNamespaceName, &configv1.Infrastructure{}, &capiv1.Cluster{}, &capav1.AWSCluster{})
+	})
+
+	Context("With a supported platform", func() {
+		Context("When there is no Core Cluster", func() {
+			BeforeEach(func() {
+				r = &CoreClusterReconciler{
+					ClusterOperatorStatusClient: operatorstatus.ClusterOperatorStatusClient{
+						Client:           cl,
+						ManagedNamespace: testNamespaceName,
+					},
+					Cluster:  &capiv1.Cluster{},
+					Infra:    openshiftInfrastructureCopy.DeepCopy(),
+					Platform: openshiftInfrastructureCopy.Status.PlatformStatus.Type,
+				}
+			})
+
+			Context("When there is no Infra Cluster", func() {
+				It("should error saying there is no infra cluster", func() {
+					By("running the core cluster controller reconciliation")
+					expectedErrors := struct {
+						infraClusterNotFound string
+						coreClusterNotFound  string
+					}{
+						infraClusterNotFound: "awsclusters.infrastructure.cluster.x-k8s.io \"test-ocp-infrastructure-name\" not found",
+						coreClusterNotFound:  "clusters.cluster.x-k8s.io \"test-ocp-infrastructure-name\" not found",
+					}
+
+					_, err := r.Reconcile(ctx, reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Namespace: testNamespaceName,
+							Name:      testInfraName,
+						},
```

Comment:
We should get rid of these now that we are setting up `startManager()` in the top level BeforeEach